docs/book/src/migration/plugin/v2_v3.md

```diff
@@ -1,3 +1,570 @@
-# Migration from `go.kubebuilder.io` v2 to v3
+# Migrating your projects to use v3+ plugins
 
-TODO
+Make sure you understand the [differences between Kubebuilder v2 and v3](/migration/v2vsv3.md)
+before continuing.
+
+Please ensure you have followed the [installation guide](/quick-start.md#installation)
+to install the required components.
+
+<aside class="note attention">
+<h1>Note</h1>
+
+The most straightforward way to upgrade your project to get all the latest changes and improvements would be to follow the [Migration Guide v2 to V3][migration-guide-v2-to-v3].
+
+</aside>
+
+## Steps to migrate
+
+<aside class="note attention">
+<h1>Note</h1>
+
+This guide only covers the breaking and most significant changes in the scaffolds. Note that the following steps
+will not migrate the API versions which are deprecated `apiextensions.k8s.io/v1beta1`, `admissionregistration.k8s.io/v1beta1`, `cert-manager.io/v1alpha2`.
+
+</aside>
+
+### Update your plugin version into the PROJECT file
+
+Please ensure you have followed the steps to upgrade your Project version to [v3][project-v3]. Then, update the `layout` setting to the new plugin version ` go.kubebuilder.io/v3` as follows:
+
+```ymal
+domain: my.domain
+layout: go.kubebuilder.io/v3
+projectName: example
+repo: example
+resources:
+- api:
+    crdVersion: v1beta
+  group: webapp
+  kind: Guestbook
+  version: v1
+  webhooks:
+    webhookVersion: v1beta
+version: 3-alpha
+```
+
+### Upgrade the Go version and its dependencies:  
+
+Ensure that your `go.mod` is using the go version `1.15` as the following dependencies versions:
+
+```
+go 1.15
+
+require (
+	github.com/go-logr/logr v0.3.0
+	github.com/onsi/ginkgo v1.14.1
+	github.com/onsi/gomega v1.10.2
+	k8s.io/api v0.19.2
+	k8s.io/apimachinery v0.19.2
+	k8s.io/client-go v0.19.2
+	sigs.k8s.io/controller-runtime v0.7.0
+)
+```
+
+### Update the golang image 
+
+In th Dockerfile, replace:
+
+```
+# Build the manager binary
+FROM golang:1.13 as builder
+```
+
+With:
+```
+# Build the manager binary
+FROM golang:1.15 as builder
+```
+
+### Update your Makefile
+
+To allow the tool scaffold the new API versions and use properly the new controller-gen, replace:
+
+```
+CRD_OPTIONS ?= "crd:trivialVersions=true"
+```
+
+With:
+
+```
+CRD_OPTIONS ?= "crd:trivialVersions=true,preserveUnknownFields=false"
+```
+
+To allow downloading the newer versions of the K8S binaries required by EnvTest in the `bin/` directory of your project instead of use the globally setup, replace:
```

Comment:
```suggestion
To allow downloading the newer versions of the K8S binaries required by EnvTest into the `bin/` directory of your project instead of the global setup, replace:
```